code/src/main/android/orxAndroidSupport.cpp

```diff
@@ -536,31 +535,36 @@ static void orxAndroid_HandleGameInput(struct android_app *_pstApp)
             stPayload.stTouch.u32ID = event->pointers[iIndex].id;
             stPayload.stTouch.fX = sstAndroid.fSurfaceScale * orxANDROID_GET_AXIS_X(event, iIndex);
             stPayload.stTouch.fY = sstAndroid.fSurfaceScale * orxANDROID_GET_AXIS_Y(event, iIndex);
-
             orxEVENT_SEND(orxEVENT_TYPE_SYSTEM, orxSYSTEM_EVENT_TOUCH_END, orxNULL, orxNULL, &stPayload);
-            sstAndroid.u32TouchCount--;
+
+            orxASSERT(stPayload.stTouch.u32ID < orxANDROID_KU32_MAX_MOTION_POINTERS);
+            sstAndroid.au32PendingMotionActions[stPayload.stTouch.u32ID] = orxFALSE;
             break;
           }
           case AMOTION_EVENT_ACTION_DOWN: /* Always first finger! */
           {
-            /** OrxMouse pressed status depends on the orxSYSTEM_EVENT_TOUCH_BEGIN and orxSYSTEM_EVENT_TOUCH_END events.
+            /** In orx, pressed status depends on the orxSYSTEM_EVENT_TOUCH_BEGIN and orxSYSTEM_EVENT_TOUCH_END events.
              * In the rare case Android fails to deliver the Up/Cancel event we must emulate the Up event.
-             * Otherwise orx would indefinitely assume that the mouse is pressed.
+             * Otherwise orx would indefinitely assume pressed state.
+             *
+             * See https://issuetracker.google.com/issues/401872146
              */
-            while(sstAndroid.u32TouchCount > 0)
+            for(iIndex = 0; iIndex < orxANDROID_KU32_MAX_MOTION_POINTERS && sstAndroid.au32PendingMotionActions[iIndex] == orxTRUE; iIndex++)

```

Comment:
Are we certain there won't be any "holes" in this array?
What if pointers 1, 2, 3 where down, then 2 went up before we hit the condition leading us here?